### PR TITLE
feat: introduces squad tracking events

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
- Introduces new tracking events for the squads beta
- Introduced a new way to track optional prev/next on stepped modals, would love to hear your thoughts. After this PR we can modify the other implementation as well.

## Events

There are a bunch of new events, without noting them down here, I refer to the taxonomy sheet:
https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=2069206456

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-951 #done

### CODE REVIEW

**Comment by idoshamun** [comment]
I haven't yet reviewed carefully but skimmed through and noticed this PR misses two new properties we added to post events
image

And also two new post events
image

They are not in the Squads tab because they belong to the posts section but we tagged you in these changes

**Comment by rebelchris** [comment]
@idoshamun Maybe we should avoid these tabs, I totally forgot about those.
And for me it doesn't make things quicker or easier.

File: packages/shared/src/lib/analytics.ts [hunk 1]
```diff
@@ -35,13 +44,26 @@ export enum AnalyticsEvent {
   ClickNotificationDismiss = 'click notification dismiss',
   DisableNotification = 'disable notification',
   // notifications - end
+  // squads - start
+  ViewSquadInvitation = 'view squad invitation',
+  ClickJoinSquad = 'click join squad',
+  CompleteJoiningSquad = 'complete joining the squad',
```
Comment (idoshamun) [inline packages/shared/src/lib/analytics.ts:50]:
```suggestion
  CompleteJoiningSquad = 'complete joining squad',
```

File: packages/webapp/pages/squads/[handle]/[token].tsx [hunk 2]
```diff
@@ -79,11 +84,37 @@ const SquadReferral = ({ token, handle }: SquadReferralProps): ReactElement => {
       },
     },
   );
+
+  useEffect(() => {
+    if (trackedImpression || !member) return;
+
+    trackEvent({
+      event_name: AnalyticsEvent.ViewSquadInvitation,
+      extra: JSON.stringify({
```
Comment (idoshamun) [inline packages/webapp/pages/squads/[handle]/[token].tsx:93]: Why don't you use `joinSquadAnalyticsExtra` here?

**Comment by idoshamun** [comment]
@rebelchris from a data analysis perspective, it's much easier to handle. But we need to find better ways to organize these changes.

**Comment by idoshamun** [comment]
The rest looks great

**Comment by idoshamun** [comment]
@rebelchris one more point, this is a tricky one. When I'm clicking "read article" on a share post, I think that current implementation will provide the details of the share post, and I think we need to provide the details post that was shared. It's very important from feed perspective as well. We can add an extra field to indicate the id of the share post so we can know that it was triggered due to this. wdyt?

File: packages/shared/src/components/ShareOptionsMenu.tsx [hunk 3]
```diff
@@ -109,14 +109,19 @@ export default function ShareOptionsMenu({
             <MenuIcon Icon={DefaultSquadIcon} />
           ),
           text: `Share to ${squad.name}`,
-          action: () =>
+          action: () => {
+            trackEvent({ event_name: AnalyticsEvent.StartShareToSquad });
```
Comment (idoshamun) [inline packages/shared/src/components/ShareOptionsMenu.tsx:113]: This is a post event and should include all the properties we include in other post events (see for example click, or upvote post)

File: packages/shared/src/components/ShareOptionsMenu.tsx [hunk 4]
```diff
@@ -109,14 +109,19 @@ export default function ShareOptionsMenu({
...
             openModal({
               type: LazyModal.PostToSquad,
               props: {
                 squad,
                 post,
+                onSharedSuccessfully: () => {
+                  trackEvent({ event_name: AnalyticsEvent.ShareToSquad });
```
Comment (idoshamun) [inline packages/shared/src/components/ShareOptionsMenu.tsx:120]: Same as above here

File: packages/shared/src/components/widgets/SocialShare.tsx [hunk 5]
```diff
@@ -61,15 +61,20 @@ export const SocialShare = ({
       }),
     );
 
-  const onShareToSquad = (squad: Squad) =>
+  const onShareToSquad = (squad: Squad) => {
+    trackEvent({ event_name: AnalyticsEvent.StartShareToSquad });
```
Comment (idoshamun) [inline packages/shared/src/components/widgets/SocialShare.tsx:65]: Same here

File: packages/shared/src/components/widgets/SocialShare.tsx [hunk 6]
```diff
@@ -61,15 +61,20 @@ export const SocialShare = ({
...
     openModal({
       type: LazyModal.PostToSquad,
       props: {
         squad,
         post,
-        onSharedSuccessfully: onSquadShare,
+        onSharedSuccessfully: () => {
+          trackEvent({ event_name: AnalyticsEvent.ShareToSquad });
```
Comment (idoshamun) [inline packages/shared/src/components/widgets/SocialShare.tsx:72]: Same here

**Review by idoshamun — APPROVED** [review]

**Pull request merged**